### PR TITLE
PTV-1773: Clean up subaction unit section of provenance

### PR DESCRIPTION
**Pull request by MrCreosote**
Make class immutable, add a builder.

Since SAs will now reject malformed URLs, I pulled all the URLs from the
production WS DB and ran them through `.withCodeURL()` and all of them
were fine.

### CODE REVIEW

File: lib/Bio/KBase/workspace/Client.pm [hunk 1]
```diff
@@ -9501,6 +9501,8 @@ Information about a subaction that is invoked by a provenance action.
         The SubAction structure allows for specifying information about SAs
```
Comment (MrCreosote) [inline lib/Bio/KBase/workspace/Client.pm:9501]: generated

File: lib/biokbase/workspace/client.py [hunk 2]
```diff
@@ -826,7 +826,8 @@ def save_objects(self, params, context=None):
            updated between a PA invoked on day T and another PA invoked on
```
Comment (MrCreosote) [inline lib/biokbase/workspace/client.py:826]: generated

File: src/us/kbase/workspace/SubAction.java [hunk 3]
```diff
@@ -29,6 +29,8 @@
  *         The SubAction structure allows for specifying information about SAs
```
Comment (MrCreosote) [inline src/us/kbase/workspace/SubAction.java:29]: generated

File: src/us/kbase/workspace/database/provenance/SubAction.java [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+			this.codeURL = Common.processURL(codeURL);
+			return this;
+		}
+
+		/** Set a URL pointing to the subaction's code, e.g. Github, Gitlab, etc.
+		 * @param codeURL the code URL. Null removes any current URL in the builder.
+		 * @return this builder.
+		 */
+		public Builder withCodeURL(final URL codeURL) {
+			// TODO PROV integration error test
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/provenance/SubAction.java:173]: More `TODO PROV`s to handle in later PRs to reduce the size of this one

**Review by ialarmedalien — APPROVED** [review]
Looks tenta-cool!

**Pull request merged**